### ISSUE TITLE
Mac OS X, watchOS, tvOS Support

### ISSUE THREAD

**Issue by cjwirth** [malcommac/SwiftDate#115]
I tried installing SwiftDate with Carthage, but only the framework for iOS was built. 

This was pretty easy to fix by adding an OS X target, and a shared scheme for it as well. However, I also had to comment out everything in `SwiftDate.h` because it was importing UIKit. Is this necessary? It seems like other Swift-only libraries don't have one.

I could pretty easily add some targets and schemes to get it to build for other OSs as well, but what are your thoughts on `SwiftDate.h`?

**Comment by cjwirth** [comment]
When I was doing this yesterday, I realized I had a super old version of Carthage installed. I'll have to try this again with the new version and see if I still have the same problems. :sweat_smile:

**Comment by Hout** [comment]
@malcommac Can you answer this one pls?

**Comment by malcommac** [comment]
Just added TvOS, OSX and WatchOS shared scheme frameworks: commit https://github.com/malcommac/SwiftDate/commit/430813365b877a26b12fe2c3067e661736c25feb.
@cjwirth can you confirm if it works fine for you?

**Comment by rcdilorenzo** [comment]
@malcommac I can't seem to get it to work for the Mac platform.

**Comment by phimage** [comment]
@rcdilorenzo  (@malcommac) the source are not added into the OSX target "Build Phase" "Compile Sources"

@malcommac I think to keep same framework names for Carthage "product name" ("Build Settings") must be edited in OSX, TVOS, WatchOS target to replace $(TARGET_NAME) by SwiftDate

**Comment by cjwirth** [comment]
I just tried it right now, and this is what I got:

```
> mkdir test && cd test

> echo 'github "malcommac/SwiftDate"' > Cartfile

> carthage version
0.11.0

> carthage update
*** Cloning SwiftDate
*** Checking out SwiftDate at "2.0.2"
*** xcodebuild output can be found in /var/folders/2b/j65n_shn79v2m2zb93r6gqcc0000gn/T/carthage-xcodebuild.kBibXZ.log
*** Building scheme "SwiftDate" in SwiftDate.xcodeproj
2016-01-06 09:51:50.362 xcodebuild[35941:2131692] ### Failed to load Addressbook class CNContactNameFormatter
2016-01-06 09:52:02.041 xcodebuild[36005:2132008] ### Failed to load Addressbook class CNContactNameFormatter

> ls Carthage/Build
iOS
```

When I added the extra target/scheme and commented out the UIKit things, it did make multiple directories in my Build directory...

**Comment by malcommac** [comment]
(on develop branch)
Just fixed the issue with compile files in OSX; it should now compile.
However SwiftDate.xcodeproj now uses CocoaPods and Nimble as framework so when I try to compile it I get: ".... ld: framework not found Nimble" (and I can't compile the xcworkspace used by cocoa pods to import Nimble).
I think the solution is to make a single independent xcproj without any reference to CocoaPods, add target to it and make another project with test files and link to nimble and cocoapods.
@Hout: do you agree?

**Comment by cjwirth** [comment]
I made some configuration changes so Quick/Nimble aren't included in the main framework, only the test target. I also got rid of some extraneous targets that didn't have to be there. Take a look at #141 and tell me what you think.

**Comment by Hout** [comment]
The change is quite big and needs investigation. I find the issue it tries to solve real though. 

@malcommac we need time to investigate and we are already in the release of v3.0.0. Shall we line this one up for v3.1.0?

**Comment by malcommac** [comment]
Go for 3.0. I'll investigate further this week to release a minor version update (3.0.1?) which enable this and carthage

**Comment by malcommac** [comment]
Just solved in 3.0.1. Thank you for your report.
If you have problems make sure to remove Carthage folder in your project in order to do a clean fetch/clone/build.
